includes/DataObjects/Request.php

```diff
@@ -256,6 +259,11 @@ public function hasComments()
 		if ($this->hasCommentsResolved) {
 			return $this->hasComments;
 		}
+        
+		if (isset(self::$commentsResults[$this->id])) {
+		    $this->hasCommentsResolved = true;
+		    return $this->hasComments = self::$commentsResults[$this->id];
+		}
```

Comment:
OK, so I'm slightly confused about this hunk of code.

If I'm right, this creates a global cache (`self::$commentsResults`) of whether the comments have been resolved, right, despite the local cache (`$this->hasComments`)?

I get the feeling that we might be better by default caching the returned object from the `getBy*()` functions so we only have to retrieve the object from the database once. At a global scale, I'm pretty sure there are some instances in the templates where we specifically do a database request direct from the template (urgh) along the lines of `{Foo::getById(4, gGetDb())->getBar()}`. Opinions @FastLizard4 ?